src/resources/MachineLearning/CaseClassificationConfiguration/CaseClassificationConfiguration.ts

```diff
@@ -1,32 +1,52 @@
 import API from '../../../APICore';
 import {New} from '../../BaseInterfaces';
 import Resource from '../../Resource';
-import {CaseClassificationConfigurationModel} from './CaseClassificationConfigurationInterfaces';
+import {
+    CaseClassificationConfigurationModel,
+    CaseClassificationContentFields,
+    CaseClassificationContentFieldsParams,
+    CaseClassificationDocumentGroupPreview,
+    CaseClassificationDocumentGroupPreviewParams,
+} from './CaseClassificationConfigurationInterfaces';
 
 export default class CaseClassificationConfiguration extends Resource {
-    static baseUrl = `/rest/organizations/${API.orgPlaceholder}/machinelearning/configuration/caseclassif/model`;
+    static baseUrl = `/rest/organizations/${API.orgPlaceholder}/machinelearning/configuration/caseclassif`;
+    static modelUrl = `${CaseClassificationConfiguration.baseUrl}/model`;
+    static fieldsUrl = `${CaseClassificationConfiguration.baseUrl}/fields`;
+    static previewUrl = `${CaseClassificationConfiguration.baseUrl}/preview`;
 
     create(configModel: New<CaseClassificationConfigurationModel, 'modelId'>) {
         return this.api.post<CaseClassificationConfigurationModel>(
-            CaseClassificationConfiguration.baseUrl,
+            CaseClassificationConfiguration.modelUrl,
             configModel
         );
     }
 
     delete(modelId: string) {
-        return this.api.delete(`${CaseClassificationConfiguration.baseUrl}/${modelId}`);
+        return this.api.delete(`${CaseClassificationConfiguration.modelUrl}/${modelId}`);
     }
 
     get(modelId: string) {
         return this.api.get<CaseClassificationConfigurationModel>(
-            `${CaseClassificationConfiguration.baseUrl}/${modelId}`
+            `${CaseClassificationConfiguration.modelUrl}/${modelId}`
         );
     }
 
     update(configModel: CaseClassificationConfigurationModel) {
         return this.api.put<CaseClassificationConfigurationModel>(
-            `${CaseClassificationConfiguration.baseUrl}/${configModel.modelId}`,
+            `${CaseClassificationConfiguration.modelUrl}/${configModel.modelId}`,
             configModel
         );
     }
+
+    fields(params: CaseClassificationContentFieldsParams) {
+        return this.api.post<CaseClassificationContentFields>(CaseClassificationConfiguration.fieldsUrl, params);
+    }
+
+    preview(params: CaseClassificationDocumentGroupPreviewParams) {
```

Comment:
The expected payload from the [Swagger docs](https://platformdev.cloud.coveo.com/docs?urls.primaryName=Machine%20Learning%20Configuration#/Case%20Classification%20Configuration/rest_organizations_paramId_machinelearning_configuration_caseclassif_preview_post) does not match the `CaseClassificationContentFieldsParams` interface.

Should the `params` have the type `CaseClassificationContentFieldsParams`?

Reply:
my bad, I got confused with the `CaseClassificationDocumentGroupPreview` interface